src/Provider/AbstractProvider.php

```diff
@@ -126,11 +125,7 @@ public function __construct(array $options = [], array $collaborators = [])
         $this->setRequestFactory($collaborators['requestFactory']);
 
         if (empty($collaborators['httpClient'])) {
-            $client_options = $this->getAllowedClientOptions($options);
-
-            $collaborators['httpClient'] = new HttpClient(
-                array_intersect_key($options, array_flip($client_options))
-            );
+            throw new \LogicException('Must specify client');
```

Comment:
The previous workflow no longer works, we cannot create a default client since we don't know what's available.

Reply:
One could check if the guzzle client class exists and instantiate if it does. 

That could be a smoother transition but it will be less generic and impartial.

Reply:
I agree, was thinking about extracting that logic into a factory, something like

```php
FallbackHttpClient::discover();
```

which could then do some heuristic to find some popular clients. WDYT?

Reply:
There are 3 ways. 
1) No fallback
2) Just Guzzle. 
3) https://github.com/php-http/discovery

To "find popular clients" is the job of the discovery package.

Reply:
OK, so we can test for any of those and try to construct the client.